implementations/rust/ockam/ockam_command/src/util/config.rs

```diff
@@ -265,25 +262,38 @@ impl OckamConfig {
         project_node_route: String,
         project_id: String,
         project_identity_id: String,
-    ) -> Result<(), ConfigError> {
+    ) -> Result<()> {
         let mut inner = self.inner.writelock_inner();
         // MultiAddr can't be serialised with serde, thus we just
         // check that the conversion is going to succeed in the
         // future, but then just pass the string value through.
         let _ = MultiAddr::from_str(&project_node_route)
             .map_err(|e| ConfigError::FailedConvert("MultiAddr".into(), e.to_string()))?;
-
+        trace! {
+            id = %project_id,
+            name = %project_name,
+            route = %project_node_route,
+            identity_id = %project_identity_id,
+            "Project stored in lookup table"
+        };
         inner.lookup.set_project(
             project_name,
             project_node_route,
             project_id,
             project_identity_id,
         );
-
         Ok(())
     }
 }
 
+impl Drop for OckamConfig {
+    fn drop(&mut self) {
+        // To reduce temporary coupling between config setters/updaters and actually
+        // updating the config file, we try to update the config file when it is dropped.
+        let _ = self.atomic_update().run();
```

Comment:
Could you explain a bit more why this is needed?

Reply:
Every time you call one of the setters/updaters you have to remember to call the `atomic_update` method. We already forgot about calling the `atomic_update` method in a couple of places, so I thought it would be nice to force calling it somehow.

For example, [this command](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/configuration/set.rs#L26) was doing nothing.

Reply:
I see. I am not sure that the `Drop` impl is the right place. For now, can you at least log the error?

Reply:
Sure, done!

Reply:
I think I've found a case where this won't work. Let me check something before merging this.